core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java

```diff
@@ -424,9 +430,12 @@ public void deleteFile(EmailMessage f) throws FileSystemException {
 	}
 	@Override
 	public EmailMessage moveFile(EmailMessage f, String destinationFolder, boolean createFolder) throws FileSystemException {
-		ExchangeService exchangeService = getConnection();
+		String folderNameToUse = getFolderNameToUse(destinationFolder);
+		String mailbox = getMailboxToUse(destinationFolder);
+		ExchangeService exchangeService = getConnection(mailbox);
+
 		try {
-			FolderId destinationFolderId = getFolderIdByFolderName(exchangeService, destinationFolder, createFolder);
+			FolderId destinationFolderId = cache.getFolder(mailbox, folderNameToUse);
```

Comment:
This assumes the folderNameToUse is in the cache. I would like to suggest to keep a call to getFolderIdByFolderName here, maybe extended with a mailbox argument. The implementation of that method can use the cache, if appropriate, but the higher level methods need not be aware of it.